src/mochijson2.erl

```diff
@@ -149,6 +149,43 @@ parse_decoder_options([{format, Format} | Rest], State)
   when Format =:= struct orelse Format =:= eep18 orelse Format =:= proplist ->
     parse_decoder_options(Rest, State#decoder{object_hook=Format}).
 
+
+-ifdef(map_available).
```

Comment:
@etrepum I thought you can't use macros in function head(s), so this was the solution I come up with, albeit not the prettiest one.

I tried the following:

```erlang
-ifdef(map_avaliable)
json_encode(M, State) when is_map(M) ->
    json_encode_map(M, State);
-endif
json_encode(...) ->
    ...
```

This is giving out to me when compiling on 18.3.